File: micro-digiservices/ecreditservice/src/main/java/io/digiservices/ecreditservice/dto/CreditDataResponse.java
package io.digiservices.ecreditservice.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class CreditDataResponse {
    // Données de petit_credit
    private String moyenPerson;
    private String bien;
    private BigDecimal capital;
    private BigDecimal creance;
    private BigDecimal dette;
    private String statutActivite;
    private String experience;
    private String lieuxAct;
    private String personEmp;
    private String lien;
    private String nombre;
    private String referenceCredit;
    private BigDecimal cumulCredit;
    private Integer nbreCredit;

    // Données de frequence
    private Integer frequenceValue;
    private LocalDateTime frequenceCreatedAt;

    // Données de garantieMat
    private String garantieLibele;
    private BigDecimal garantieMontant;
    private LocalDateTime garantieCreatedAt;

    // Données de localisation
    private String itAss;
    private String itPc;

    // Données agrégées en JSON
    private String produitsData;  // JSON string
    private String chargesData;   // JSON string
    private String cautionsData;  // JSON string
}

File: frontend/ultima-ng-19.0.0/src/app/interface/credit.data.response.ts
export interface CreditDataResponse {
    // Données de petit_credit
    moyenPerson?: string;
    bien?: string;
    capital?: number;
    creance?: number;
    dette?: number;
    statutActivite?: string;
    experience?: string;
    lieuxAct?: string;
    personEmp?: string;
    lien?: string;
    nombre?: string;
    referenceCredit?: string;
    cumulCredit?: number;
    nbreCredit?: number;

    // Données de frequence
    frequenceValue?: number;
    frequenceCreatedAt?: string; // ISO string format: "2024-12-27T10:30:00"

    // Données de garantieMat
    garantieLibele?: string;
    garantieMontant?: number;
    garantieCreatedAt?: string; // ISO string format: "2024-12-27T10:30:00"

    // Données de localisation
    itAss?: string;
    itPc?: string;

    // Données agrégées en JSON
    produitsData?: string; // JSON string
    chargesData?: string; // JSON string
    cautionsData?: string; // JSON string
}
